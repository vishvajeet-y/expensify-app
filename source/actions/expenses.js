import { v4 as uuidv4 } from 'uuid'
import database from '../Firebase/firebase'

// export const addExpense=({  //destructurng payload and asssigning default value
//     description='',
//     note='',
//     ammount=0,
//     createAt=0
// }={})=>({
//     type:'ADD_EXPENSE',
//     expense:{
//     id:uuidv4(),
//     description,
//     note,
//     ammount,
//     createAt
//     }
// })

export const addExpense=(expense)=>{
    return {
        type:'ADD_EXPENSE',
        expense
    }
}

export const startAddExpense=(expenseData={})=>{
    return (dispatch)=>{
        const {
            description='',
            note='',
            createAt=0,
            ammount=0
        }=expenseData

        const expense={description,note,ammount,createAt}

      return  database.ref('expenses').push(expense).then((ref)=>{
            dispatch(addExpense({
                id:ref.key,
                ...expense
            }))
        })
    }
}

//Remove Expense
export const removeExpense=({id}={})=>({
    type:'REMOVE_EXPENSE',
    id
})

//Edit Expense
export const editExpense=(id,updates)=>({
   type:'EDIT_EXPENSE',
   id,updates
})
