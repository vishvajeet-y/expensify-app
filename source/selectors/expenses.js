import moment from 'moment'
const getVisibleExpenses=(expenses,{text,sortBy,startDate,endDate})=>{
    return expenses.filter((expense)=>{
        const createAtMoment=moment(expense.createAt)
        const startDateMatch=startDate?startDate.isSameOrBefore(createAtMoment) :true
        const endDateMatch=endDate ?endDate.isSameOrAfter(createAtMoment) :true
        const textMatch=expense.description.toLowerCase().includes(text.toLowerCase())
        return startDateMatch&&endDateMatch&&textMatch
    }).sort((a,b)=>{
        if(sortBy=='date')
        return a.createAt<b.createAt?1:-1
        if(sortBy=='ammount')
        return a.ammount<b.ammount ?1:-1
    })
   }
   export default getVisibleExpenses