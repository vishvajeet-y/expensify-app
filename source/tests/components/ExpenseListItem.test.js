import React from 'react'
import ExpenseListItem from '../../components/ExpenseList-item'
import {shallow} from 'enzyme'
import expenses from '../fixture/expenses'

test('should render expenseList Item',()=>{
    const wrapper=shallow(<ExpenseListItem {...expenses[0]}/>)
    expect(wrapper).toMatchSnapshot()
})
