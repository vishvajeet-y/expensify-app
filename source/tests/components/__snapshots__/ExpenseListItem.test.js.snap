// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenseList Item 1`] = `
<div>
  <Link
    to="./edit/1"
  >
     
    <h3>
      GUM
    </h3>
  </Link>
  <p>
    Ammount: 
    â‚¹10.00
     
  </p>
  <p>
     Date :
    1st January,1970
  </p>
</div>
`;
